src/rpc/handlers/NFTOffers.cpp

```diff
@@ -126,6 +123,7 @@ enumerateNFTOffers(
     for (auto const& offer : offers)
         appendNftOfferJson(offer, jsonOffers);
```

Comment:
Why don't we just return the json offer object from this function call, and push the value into the array? I feel like passing `jsonOffers` into the function is an unecessary amount of indirection and side effects, with no real benefit

Reply:
i agree! i have now done it slightly differently but the idea remains the same.